cpp/src/copying/scatter.cu

```diff
@@ -329,6 +332,104 @@ std::vector<std::unique_ptr<table>> scatter_to_tables(
     input, partition_map, mr, stream);
 }
 
+std::unique_ptr<column> boolean_mask_scatter(
+    column_view const& source, column_view const& target,
+    column_view const& boolean_mask,
+    rmm::mr::device_memory_resource *mr,
+    cudaStream_t stream) {
+
+    auto indices = cudf::make_numeric_column(data_type{INT32},
+                                  target.size(), UNALLOCATED, stream, mr);
+    auto mutable_indices = indices->mutable_view();
+
+    thrust::sequence(rmm::exec_policy(stream)->on(stream),
```

Comment:
Isn't there a way to avoid materializing this sequence? Could you somehow use a counting iterator as input to apply_boolean_mask?

Reply:
It expects a table_view, so need a column.

Reply:
Will do it in a follow-up PR, else this might PR would burst.

Reply:
I didn't mean in this PR.